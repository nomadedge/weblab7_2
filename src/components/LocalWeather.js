import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchLocalWeather } from '../actions/LocalWeatherActions';

import Loader from './Loader';
import CityWeather from './CityWeather';

import '../styles/Weather.css';

class LocalWeather extends Component {
    componentDidMount() {
        this.props.fetchLocalWeather();
    }

    render() {
        const { localWeather, fetchLocalWeather } = this.props;

        if (localWeather.isFetching) {
            return (
                <>
                    <button onClick={fetchLocalWeather}>Update local weather</button>
                    <div id='weather-container'>
                        <Loader />
                    </div>
                </>
            )
        }
        if (localWeather.error !== null) {
            return (
                <>
                    <button onClick={fetchLocalWeather}>Update local weather</button>
                    <div id='weather-container'>
                        <div>{localWeather.error}</div>
                    </div>
                </>
            )
        }
        else {
            return (
                <>
                    <button onClick={fetchLocalWeather}>Update local weather</button>
                    <div id='weather-container'>
                        <CityWeather cityWeather={localWeather} />
                    </div>
                </>
            );
        }
    }
}

const mapStateToProps = ({ localWeather }) => {
    return {
        localWeather: localWeather,
    };
};
const mapDispatchToProps = {
    fetchLocalWeather: fetchLocalWeather
};

export default connect(mapStateToProps, mapDispatchToProps)(LocalWeather);