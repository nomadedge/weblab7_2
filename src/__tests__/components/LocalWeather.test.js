import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import LocalWeather from '../../components/LocalWeather';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('LocalWeather component', () => {
    test('matches the snapshot on loading', () => {
        const localCity = {
            isFetching: true,
            city: null,
            weather: {},
            error: null
        };

        const store = mockStore({
            localWeather: localCity
        });

        const tree = renderer.create(
            <Provider store={store}>
                <LocalWeather />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot on custom error', () => {
        const localCity = {
            isFetching: false,
            city: 'Kiyv',
            weather: {},
            error: 'Weather for your city is not available :('
        };

        const store = mockStore({
            localWeather: localCity
        });

        const tree = renderer.create(
            <Provider store={store}>
                <LocalWeather />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot on success', () => {
        const localCity = {
            isFetching: false,
            city: 'Kiyv',
            weather: {
                iconUrl: "http://openweathermap.org/img/w/50n.png",
                description: "mist",
                temperature: "14",
                pressure: 1022,
                humidity: 100,
                windSpeed: 3.6
            },
            error: null
        };

        const store = mockStore({
            localWeather: localCity
        });

        const tree = renderer.create(
            <Provider store={store}>
                <LocalWeather />
            </Provider>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});