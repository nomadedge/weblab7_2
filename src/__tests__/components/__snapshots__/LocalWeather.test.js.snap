// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalWeather component matches the snapshot on error 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Update local weather
  </button>,
  <div
    id="weather-container"
  >
    <div>
      Weather for your city is not available :(
    </div>
  </div>,
]
`;

exports[`LocalWeather component matches the snapshot on loading 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Update local weather
  </button>,
  <div
    id="weather-container"
  >
    <div
      id="spinner"
    />
  </div>,
]
`;

exports[`LocalWeather component matches the snapshot on success 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Update local weather
  </button>,
  <div
    id="weather-container"
  >
    <div
      id="left-block"
    >
      <div
        id="city-name"
      />
      <div
        id="icon-temp"
      >
        <img
          alt="Weather icon"
          src="http://openweathermap.org/img/w/50n.png"
        />
        <div
          id="temp"
        >
          14
          Â°C
        </div>
      </div>
    </div>
    <div
      id="right-block"
    >
      <div
        className="info"
      >
        <div>
          Description:
        </div>
        <div
          className="value"
        >
          mist
        </div>
      </div>
      <div
        className="info"
      >
        <div>
          Barometer:
        </div>
        <div
          className="value"
        >
          1022
           hPa
        </div>
      </div>
      <div
        className="info"
      >
        <div>
          Humidity:
        </div>
        <div
          className="value"
        >
          100
           %
        </div>
      </div>
      <div
        className="info"
      >
        <div>
          Wind:
        </div>
        <div
          className="value"
        >
          3.6
           m/s
        </div>
      </div>
    </div>
  </div>,
]
`;
